### PR TITLE
#7 fastforward 280 commits to contenta_json_api version 1.449

### PR DESCRIPTION
**Pull request by martinfrances107**
Just updating contenta_json_api

### CODE REVIEW

**Comment by e0ipso** [comment]
What if we don't hardcode a tag at all and we get the latest:

```
# Get new tags from remote
git fetch --tags

# Get latest tag name
latestTag=$(git describe --tags `git rev-list --tags --max-count=1`)

# Checkout latest tag
git checkout $latestTag
```

**Review by e0ipso — CHANGES_REQUESTED** [review]
Please check out the latest tag instead of a hard-coded one.

**Comment by e0ipso** [comment]
@martinfrances107 thanks a lot for your work in this repo!

**Comment by martinfrances107** [comment]
Checking out the latest tag is a good suggestion

Although in some environments building the same docker image,  say 1000 days apart, and getting different results is a no no. 

So talking cavalierly  in 1000 days when drupal8 bumps its min php version to 71 we will have a broken image ... or whatever hidden coupling we find between contenta_json and the docker file we will find things will break..

But in every life a little rain must fall ... and I am happy to do this and accept that thing drift slowly and have to be maintained.

@e0ipso what is you prefered way of working I have pushed the changes to 

https://github.com/martinfrances107/contenta_docker

Should I create another PR?

**Comment by martinfrances107** [comment]
Justing posting the results of the last command in the README

In the existing branch the command has alway had problems and to get the admin one time 
web page we alway needed a little manual intervention... BUT the send mail error is new 

I suggest we correct that in a follow up



docker exec -it contentadocker_php_1 init-drupal

Generating RSA private key, 2048 bit long modulus
...........................+++
...................................+++
e is 65537 (0x10001)
writing RSA key
You are about to DROP all tables in your 'contenta' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the [ok]
--notify global option.
sh: /usr/sbin/sendmail: No such file or directory
Installation complete.                                               [ok]
Unable to send email. Contact the site administrator if the problem  [error]
persists.
Recipe magazin installed                                             [status]
Congratulations, you installed Contenta JSON API!                    [status]

File: php/Dockerfile [hunk 1]
```diff
@@ -69,9 +69,11 @@ RUN mkdir -p /var/www && \
     chmod +x /usr/local/bin/init-drupal
 
 RUN cd /usr/local/src && \
-    git clone https://github.com/contentacms/contenta_jsonapi && \    
+    git clone https://github.com/contentacms/contenta_jsonapi && \
     cd /usr/local/src/contenta_jsonapi && \
-    git checkout 836683de603741e9e3beebac1b047e34c684a69c && \
+    git fetch --tags  && \
+    latestTag=$(git describe --tags `git rev-list --tags --max-count=1`) # Get latest tag name && \
```
Comment (e0ipso) [inline php/Dockerfile]: I think the comment (including the `#`) is involuntarily commenting `&& \` at the end of the line. Can we move it to a new line instead?

**Review by e0ipso — CHANGES_REQUESTED** [review]
I'd also like to address the `sendmail` regression in the same PR as well.

**Comment by e0ipso** [comment]
> Although in some environments building the same docker image, say 1000 days apart, and getting different results is a no no.

I totally agree with that. But that applies to site images, not contributed images. We need to make sure that people get the latest stable version regardless of the install method. Docker is a one of those install mehtods.

**Comment by martinfrances107** [comment]
> I'd also like to address the sendmail regression in the same PR as well.

I have just gone back to basics and see that I falsely reported that sendmail issue was new.

I have opened up a new issue so we an separate that out.

 #10

**Comment by e0ipso** [comment]
@martinfrances107 so the `sendmail` issue exists in the current `master`, right?

Also, please see the review I left on the code of this PR.

**Comment by martinfrances107** [comment]
> so the sendmail issue exists in the current master, right?

Yes.

I saw the changes in the review, but I was holding back .. because there is a question that remains in the associated issue #7 

@pcambra has suggested just pulling master, instead of "latestTag" so that the build is developer friendly. I had left a response seeking further comments... I was going to just do what the average of the crowd wanted.

**Comment by e0ipso** [comment]
@martinfrances107 `master` is fine.

**Comment by martinfrances107** [comment]
> master is fine.

Ok I have simplified by just deleting the git checkout command.

I have tested and everything looks ok

I had to do a git push -f ... and this PR did not update ... but the changes are in master of 

martinfrances107/contenta_docker

Should I create a new pull request?

PS I am updating the issue with the decision about just going with master.

**Pull request merged**